File: Introductory Problems/cpp/WeirdAlgorithm.cpp
#include <bits/stdc++.h>
using namespace std;
#define ll long long
 
void weirdAlgorithm(ll n) {
    while (true) {
        cout << n << " ";
        // base case 
        if (n == 1) {
            break;
        }
        // weird algorithm
        n = (n%2 == 0) ? n/2 : (n*3)+1;
    }
}
 
// main function
int main() {
    ios_base::sync_with_stdio(false);
    cin.tie(0);
 
    ll n;
    // number
    cin >> n;
 
    // solve
    weirdAlgorithm(n);
    
    return 0;
}

File: Introductory Problems/java/WeirdAlgorithm.java
import java.util.Scanner;

public class WeirdAlgorithm {

    private static void weirdAlgorithm(long n) {
        // print n
        System.out.print(n);
 
        // weird algorithm
        while (n != 1) {
            n = (n % 2 == 0) ? n / 2 : (n * 3) + 1;
            System.out.print(" " + n);
        }

        return;
    }

    public static void main(String[] args) {
        // scanner
        Scanner sc = new Scanner(System.in);

        long n = sc.nextLong();
        sc.close();

        // solve
        weirdAlgorithm(n);

        return;
    }
}
